src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -409,6 +409,8 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
 
             $transitions = array();
             foreach ($workflow['transitions'] as $transitionName => $transition) {
+                $transitionName = $transition['name'] ?: $transitionName;
```

Comment:
I think we should normalise the values in the `Configuration` class already to ensure that an element with the `name` key is set.